.github/workflows/main.yaml

```diff
@@ -14,18 +12,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, ubuntu-24.04, macos-12, macos-13, macos-14]
-        python: ["3.9", "3.10", "3.11", "3.12"]
-        exclude:
-          - os: macos-12
-            python: "3.11"
-          - os: macos-12
-            python: "3.12"
-          - os: macos-13
-            python: "3.11"
-          - os: macos-13
-            python: "3.12"
-
```

Comment:
What made removing the os matrix necessary?

Reply:
mostly to reduce the amount of CI resources being burnt while testing

Reply:
mm, yeah i guess that's fair, but I think we need a decent spread.
We can loose some versions of python - now needs >3.10. We probably need windows tested as well tho. macos-12 is being discontinued soon, so that can go.

Not too worried about runner cost, but we should cover as much target user arch as we can. I can always set up some specific runners if the wait for runner availability is too long after that